ecs-agent/acs/session/task_manifest_responder_test.go

```diff
@@ -0,0 +1,228 @@
+//go:build unit
+// +build unit
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package session
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	mock_session "github.com/aws/amazon-ecs-agent/ecs-agent/acs/session/mocks"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/session/testconst"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/metrics"
+	mock_metrics "github.com/aws/amazon-ecs-agent/ecs-agent/metrics/mocks"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/wsclient"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	StartingLatestTaskManifestSequenceNumber = 15
+)
+
+// setupTestManifestMessage initializes a dummy TaskManifestMessage for testing
+func setupTestManifestMessage() *ecsacs.TaskManifestMessage {
+	return &ecsacs.TaskManifestMessage{
+		ClusterArn:           aws.String(testconst.ClusterName),
+		ContainerInstanceArn: aws.String(testconst.ContainerInstanceARN),
+		MessageId:            aws.String(testconst.MessageID),
+		Tasks: []*ecsacs.TaskIdentifier{
+			{DesiredStatus: aws.String("STOPPED"), TaskArn: aws.String(testconst.TaskARN)},
+		},
+		Timeline: aws.Int64(StartingLatestTaskManifestSequenceNumber),
+	}
+}
+
+// setupTestTaskManifestResponder initializes the TaskManifestResponder for testing.
+func setupTestTaskManifestResponder(t *testing.T, factory metrics.EntryFactory, f wsclient.RespondFunc) *taskManifestResponder {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	// Create task manifest responder.
+	mockTC := mock_session.NewMockTaskComparer(ctrl)
+	mmIDA := mock_session.NewMockManifestMessageIDAccessor(ctrl)
+	mockSNA := mock_session.NewMockSequenceNumberAccessor(ctrl)
+
+	mmIDA.EXPECT().SetMessageID("").Times(1)
+	mmIDA.SetMessageID("")
+
+	mockSNA.EXPECT().SetLatestSequenceNumber(gomock.Any()).MinTimes(1)
+	mockSNA.SetLatestSequenceNumber(setupTestManifestMessage())
+
+	testTaskManifestResponder := NewTaskManifestResponder(mockTC, mockSNA, mmIDA, factory, f)
+	return testTaskManifestResponder
```

Comment:
Can we update this function to return all the mock variables we might need in a test? That way we won't need to do type assertions such as on L90 for example.